src/rust/engine/sharded_lmdb/src/lib.rs

```diff
@@ -341,40 +350,89 @@ impl ShardedLmdb {
       .await
   }
 
+  ///
+  /// Singular form of `Self::store_bytes_batch`. When storing more than one item in parallel,
+  /// prefer `Self::store_bytes_batch`.
```

Comment:
Given how this function simply delegates to `store_bytes_batch`, have you considered removing this and changing `store_bytes` to the batch API? Ack it's not as ergonomic, but to remove the risk of someone doing the wrong thing.

Or how about renaming to `store_bytes_unbatched`?